src/pyuvdata/utils/uvcalibrate.py

```diff
@@ -3,24 +3,163 @@
 # Licensed under the 2-clause BSD License
 """Code to apply calibration solutions to visibility data."""
 import warnings
+from typing import Literal
 
 import numpy as np
 
 from .pol import POL_TO_FEED_DICT, jnum2str, parse_jpolstr, polnum2str, polstr2num
 
 
+def _get_pol_conventions(
+    uvdata,
+    uvcal,
+    undo: bool,
+    uvc_pol_convention: Literal["sum", "avg"] | None,
+    uvd_pol_convention: Literal["sum", "avg"] | None,
+):
+    if uvc_pol_convention is None and uvcal.pol_convention is None:
+        warnings.warn(
+            message=(
+                "pol_convention is not specified on the UVCal object, and "
```

Comment:
I don't know that I'd want this to error when the pol convention is not specified on either UVCal and UVData, at least not if we're treating `pol_convention` as an optional parameter. Why can't this just stay a warning?

Reply:
Sorry @kartographer, I'm not following -- this is a warning?

Reply:
@steven-murray -- sorry, I mean a general `UserWarning`, not a `DeprecationWarning`. I.e., this should alert the user that there's a potential for mismatch, but I don't think this should throw an error.

Reply:
OK, I updated all the DeprecationWarning's to UserWarnings. Originally I had intended these to be DeprecationWarnings because we want to push people to use the keywords. However, it would be really annoying to have that warning if you really just don't care about the final units (like, you're just using calibration solutions to multiply your data by 2...)